### PR TITLE
Don't use XSEDE username when TAS API fails

### PR DESCRIPTION
**Pull request by julianpistorius**
Problem: When TAS API fails (for any reason) we use the XSEDE username instead of the TACC username.

Solution: Allow `jetstream.allocation.get_tacc_username` to return `None` and handle it properly.

## Description

Please describe your pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. When linking to an issue, please use `refs #...` in the description of the pull request.

## Checklist before merging Pull Requests
- [ ] New test(s) included to reproduce the bug/verify the feature
- [ ] Documentation created/updated at [Example link to documentation](https://example.test/doc#new_section) to give context to the feature
- [x] Reviewed and approved by at least one other contributor.

### CODE REVIEW

File: dev_requirements.in [hunk 1]
```diff
@@ -16,3 +16,5 @@ pip-tools
 wheel
 vcrpy
 json_delta
+freezegun
+pathlib
```
Comment (lenards) [inline dev_requirements.in]: Cool finds.

File: jetstream/test_tas_api.py [hunk 2]
```diff
@@ -28,9 +29,9 @@ def test_validate_account(self, cassette):
         """Test for a valid account based on the business logic assigned by Jetstream"""
         from jetstream.plugins.auth.validation import XsedeProjectRequired
         jetstream_auth_plugin = XsedeProjectRequired()
-        mock_user = Mock()
-        mock_user.username = 'sgregory'
-        is_jetstream_valid = jetstream_auth_plugin.validate_user(mock_user)
+        user = UserFactory.create(username='sgregory')
+        with freezegun.freeze_time('2016-09-15T05:00:00Z'):
```
Comment (steve-gregory) [inline jetstream/test_tas_api.py]: @julianpistorius care to explain why we use freezegun in the test?

File: jetstream/allocation.py [hunk 3]
```diff
@@ -59,8 +59,9 @@ def get_tacc_username(self, user):
                 user.username)
         except:
             logger.info("User: %s has no tacc username" % user.username)
-            tacc_user = user.username
-        self.username_map[user.username] = tacc_user
+            tacc_user = None
+        else:
+            self.username_map[user.username] = tacc_user
```
Comment (steve-gregory) [inline jetstream/allocation.py]: 👍

File: jetstream/allocation.py [hunk 4]
```diff
@@ -285,6 +286,9 @@ def get_or_create_allocation_source(api_allocation, update_source=False):
 def find_user_allocation_source_for(driver, user):
     tacc_user = driver.get_tacc_username(user)
     # allocations = driver.find_allocations_for(tacc_user)
+    if not tacc_user:
+        return []
+
```
Comment (lenards) [inline jetstream/allocation.py]: Okay, so we'll return an empty list when we've encountered an XSEDE to TACC lookup. 

Looks good.

File: jetstream/test_tas_api.py [hunk 5]
```diff
@@ -28,9 +29,9 @@ def test_validate_account(self, cassette):
...
+            is_jetstream_valid = jetstream_auth_plugin.validate_user(user)
```
Comment (lenards) [inline jetstream/test_tas_api.py]: Nice 👍

**Comment by lenards** [comment]
👍 :shipit:

**Pull request merged**